src/data/nat/squarefree.lean

```diff
@@ -513,7 +513,7 @@ end
 
 /-- Evaluates the `squarefree` predicate on naturals. -/
 @[norm_num] meta def eval_squarefree : expr → tactic (expr × expr)
-| `(squarefree (%%e : ℕ)) := do
+| `(@squarefree ℕ _ %%e) := do
```

Comment:
Pattern matching on the instance and attempting to unify it with the canonical one seems like a way to resolve the test-case I provided. If you think it's not worth it, please just add a TODO comment and a commented out version of the test-case.

bors d+